components/dashboard/src/workspaces/ConnectToSSHModal.tsx

```diff
@@ -55,39 +55,46 @@ interface SSHProps {
 }
 
 function SSHView(props: SSHProps) {
-    const sshCommand = `ssh '${props.workspaceId}#${props.ownerToken}@${props.ideUrl.replace(
-        props.workspaceId,
-        props.workspaceId + ".ssh",
-    )}'`;
+    const [hasSSHKey, setHasSSHKey] = useState(false);
+
+    useEffect(() => {
+        getGitpodService()
+            .server.hasSSHPublicKey()
+            .then((d) => {
+                setHasSSHKey(d);
+            })
+            .catch(console.error);
+    }, []);
+
+    const host = props.ideUrl.replace(props.workspaceId, props.workspaceId + ".ssh");
+    const sshPswCommand = `ssh '${props.workspaceId}#${props.ownerToken}@${host}'`;
+    const sshKeyCommand = `ssh '${props.workspaceId}@${host}'`;
+
     return (
-        <div className="border-t border-b border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 py-6">
-            <div className="mt-1 mb-4">
-                <AlertBox>
-                    <p className="text-red-500 whitespace-normal text-base">
-                        <b>Anyone</b> on the internet with this command can access the running workspace. The command
-                        includes a generated access token that resets on every workspace restart.
-                    </p>
-                </AlertBox>
-                <InfoBox className="mt-4">
-                    <p className="text-gray-500 whitespace-normal text-base">
-                        Before connecting via SSH, make sure you have an existing SSH private key on your machine. You
-                        can create one using&nbsp;
-                        <a
-                            href="https://en.wikipedia.org/wiki/Ssh-keygen"
-                            target="_blank"
-                            rel="noopener noreferrer"
-                            className="gp-link"
-                        >
-                            ssh-keygen
-                        </a>
-                        .
-                    </p>
-                </InfoBox>
-                <p className="mt-4 text-gray-500 whitespace-normal text-base">
+        <div>
+            <div className="space-y-4">
+                <Alert type="warning" className="whitespace-normal">
+                    <b>Anyone</b> on the internet with this command can access the running workspace. The command
+                    includes a generated access token that resets on every workspace restart.
+                </Alert>
+                <Alert type="info" className="whitespace-normal">
+                    Before connecting via SSH, make sure you have an existing SSH private key on your machine. You can
+                    create one using&nbsp;
```

Comment:
Display different `warning` `info` content for `sshPswCommand` and `sshKeyCommand`. cc @gtsiolis @loujaybee